Source/Diagnostics/Diagnostics.cpp

```diff
@@ -84,9 +84,77 @@ Diagnostics::BaseReadParameters ()
     // If user requests to plot proc_number for a serial run,
     // delete proc_number from fields_to_plot
     if (amrex::ParallelDescriptor::NProcs() == 1){
-        m_varnames.erase(
-            std::remove(m_varnames.begin(), m_varnames.end(), "proc_number"),
-            m_varnames.end());
+        m_varnames_fields.erase(
+            std::remove(m_varnames_fields.begin(), m_varnames_fields.end(), "proc_number"),
+            m_varnames_fields.end());
+    }
+
+    // Get names of particle quantities to average at each grid point
+    bool pfield_varnames_specified = pp_diag_name.queryarr("particle_fields_to_plot", m_pfield_varnames);
+    if (!pfield_varnames_specified){
+        m_pfield_varnames = {};
+    }
+#ifdef WARPX_DIM_RZ
+    if (m_pfield_varnames.size() != 0) {
+        amrex::Abort("Input error: cannot use particle_fields_to_plot not implemented for RZ");
+    }
+#endif
+
+
+    // Get parser strings for particle fields and generate map of parsers
+    bool pfield_parser_specified = false;
```

Comment:
I think I might be missing this, where/how is this variable `pfield_parser_specified` used here?

Reply:
Good catch, thanks. I'll remove it.